python_bindings/tests/engine_test.py

```diff
@@ -85,3 +91,102 @@ def test_game_engine_send_batch_returns_failures_when_communicator_closed(
         expected = requests[idx].GameID
         actual = resp.GameID()
         assert actual == expected
+
+
+def test_game_engine_send_get_remaining_tiles_batch_returns_remaining_tiles(
+    tmp_path: Path
+) -> None:
+    t1 = tiletemplates.MonasteryWithSingleRoad()
+    t2 = tiletemplates.RoadsTurn()
+    tiles = [t1, t2, t1]
+    tile_set = TileSet(
+        StartingTile=tiletemplates.SingleCityEdgeStraightRoads(),
+        Tiles=Slice_tiles_Tile(tiles),
+    )
+
+    engine = StartGameEngine(1, str(tmp_path))
+    game = engine.GenerateGame(tile_set)
+
+    request = GetRemainingTilesRequest(BaseGameID=game.ID)
+    resp, = engine.SendGetRemainingTilesBatch(
+        Slice_Ptr_engine_GetRemainingTilesRequest([request])
+    )
+    resp.Err()  # the binding would raise if this returned an error
+    engine.Shutdown()
+
+    assert resp.Denominator == len(tiles)
```

Comment:
What is the reason to go with such method of displaying probabilities, instead of simple floating point number?

Reply:
The ability to represent a fraction exactly. If it's useless, I can divide each value instead.